### PR TITLE
datapath,daemon: Fix initialization panics when IPv6 is enabled

### LINKED ISSUE #12188: 1.8-rc3 crashloop with ipv6 enabled
## Bug report

On a fresh install of 1.8-rc3 with ipv6 enabled, cilium-agents goes into a crashloop. logs provide some warnings and then a fatal message about failing to determine IPv6 for the host interface

```
level=fatal msg="Failed to initialize NodePort addrs" error="Failed to determine IPv6 of enp0s8 for NodePort" subsys=daemon
```

Deployed with `helm template cilium/cilium --version 1.8.0-rc3 --namespace=kube-system --set global.ipv4.enabled=true --set global.ipv6.enabled=true > cilium-1.8-rc3.yaml`

Some additional level=warning messages precede the fatal log are included below

```
level=warning msg="Device with a default route cannot be found for BPF NodePort device detection" error="Default route not found for family 10" subsys=daemon
level=info msg="Using auto-derived devices for BPF node port" devices="[enp0s8]" directRoutingDevice=enp0s8 subsys=daemon
level=warning msg="Session affinity for host reachable services needs kernel 5.7.0 or newer to work properly when accessed from inside cluster: the same service endpoint will be selected from all network namespaces on the host." subsys=daemon
```

**General Information**

- Cilium version (run `cilium version`)
 1.8-rc3
- Kernel version (run `uname -a`) 
Linux agent1 5.3.0-51-generic #44-Ubuntu SMP Wed Apr 22 21:09:44 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
- Orchestration system version in use (e.g. `kubectl version`, Mesos, ...)
v1.18.4+k3s1 
- Link to relevant artifacts (policies, deployments scripts, ...)
- Generate and upload a system zip:
```
curl -sLO https://git.io/cilium-sysdump-latest.zip && python cilium-sysdump-latest.zip
```

**How to reproduce the issue**

1. Deploy 1.8-rc3 with IPv6 enabled

### PR DESCRIPTION
**Pull request by brb**
This PR fixes two panics mentioned in #12188 (see commit msgs).

Fix #12188

### CODE REVIEW

**Comment by brb** [comment]
test-me-please

**Review by borkmann — APPROVED** [review]

File: daemon/cmd/kube_proxy_replacement.go [hunk 1]
```diff
@@ -264,7 +264,19 @@ func detectDevicesForNodePortAndHostFirewall(strict bool) {
...
-func finishKubeProxyReplacementInit() {
+func finishKubeProxyReplacementInit(isKubeProxyReplacementStrict bool) {
+	if option.Config.EnableNodePort {
+		if err := node.InitNodePortAddrs(option.Config.Devices); err != nil {
+			msg := "Failed to initialize NodePort addrs."
+			if isKubeProxyReplacementStrict {
+				log.WithError(err).Fatal(msg)
+			} else {
+				disableNodePort()
+				log.WithError(err).Warn(msg + " Disabling BPF NodePort.")
```
Comment (aanm) [inline daemon/cmd/kube_proxy_replacement.go:275]: Just to double check, will this show up under `cilium status`?

**Review by aanm — CHANGES_REQUESTED** [review]

File: daemon/cmd/kube_proxy_replacement.go (hunk 1 above)
Comment (brb) [reply daemon/cmd/kube_proxy_replacement.go:275]: Yes.

**Review by aanm — APPROVED** [review]

**Comment by borkmann** [comment]
provision fail on net-next suite:
```
09:11:58  0%...10%...20%...30%...40%...
09:11:58  Progress state: VBOX_E_INVALID_OBJECT_STATE
09:11:58  VBoxManage: error: Machine delete failed
09:11:58  VBoxManage: error: Medium '/root/VirtualBox VMs/ubuntu-18.04.4-amd64-59_1592555301695_52087/ubuntu-18.04.4-amd64-59-disk001.vmdk' is locked for reading by another task
09:11:58  VBoxManage: error: Details: code VBOX_E_INVALID_OBJECT_STATE (0x80bb0007), component MediumWrap, interface IMedium
09:11:58  VBoxManage: error: Context: "RTEXITCODE handleUnregisterVM(HandlerArg*)" at line 162 of file VBoxManageMisc.cpp
09:11:58  0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
09:11:58  0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
09:11:58  0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
09:11:58  0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
09:11:58  0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
09:12:00  0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
09:12:00  Clean docker images from registry
09:12:01  registry
```

**Comment by borkmann** [comment]
retest-net-next

**Review by rolinh — APPROVED** [review]
This PR with also the fix in #12198 fix the panic I reported [here](https://github.com/cilium/cilium/issues/12188#issuecomment-646278146).
However, I uncovered another bug that prevents an IPv6 only cluster to run (not a regression, already the case for v1.7.5): #12201.

I haven't tested a dual-stack cluster yet.

**Pull request merged**